server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -459,24 +598,354 @@ else if (ownerDto.getKey() != null) {
+        if (dto.getCalculatedAttributes() != null) {
+            if (dto.getCalculatedAttributes().isEmpty()) {
+                entity.setCalculatedAttributes(Collections.<String, String>emptyMap());
+            }
+            else {
+                entity.setCalculatedAttributes(dto.getCalculatedAttributes());
+            }
         }
 
-        return product;
+        if (dto.getProductAttributes() != null) {
+            if (dto.getProductAttributes().isEmpty()) {
+                entity.setProductAttributes(Collections.<String, String>emptyMap());
+            }
+            else {
+                entity.setProductAttributes(dto.getProductAttributes());
+            }
+        }
+
+        if (dto.getDerivedProductAttributes() != null) {
+            if (dto.getDerivedProductAttributes().isEmpty()) {
+                entity.setDerivedProductAttributes(Collections.<String, String>emptyMap());
+            }
+            else {
+                entity.setDerivedProductAttributes(dto.getDerivedProductAttributes());
+            }
+        }
+
+        if (dto.getBranding() != null) {
+            if (dto.getBranding().isEmpty()) {
+                entity.setBranding(Collections.<Branding>emptySet());
+            }
+            else {
+                Set<Branding> branding = new HashSet<Branding>();
+                for (BrandingDTO brandingDTO : dto.getBranding()) {
+                    if (brandingDTO != null) {
+                        branding.add(new Branding(
+                            brandingDTO.getProductId(),
+                            brandingDTO.getType(),
+                            brandingDTO.getName()));
+                    }
+                }
+                entity.setBranding(branding);
+            }
+        }
+
+        if (dto.getProvidedProducts() != null) {
+            if (dto.getProvidedProducts().isEmpty()) {
+                entity.setProvidedProductDtos(Collections.<ProvidedProduct>emptySet());
+            }
+            else {
+                Set<ProvidedProduct> products = new HashSet<ProvidedProduct>();
+                for (PoolDTO.ProvidedProductDTO providedProductDTO : dto.getProvidedProducts()) {
+                    if (providedProductDTO != null) {
+                        ProvidedProduct newProd = new ProvidedProduct();
+                        newProd.setProductId(providedProductDTO.getProductId());
+                        newProd.setProductName(providedProductDTO.getProductName());
+                        products.add(newProd);
+                    }
+                }
+                // We are using the entity's setProvidedProductDtos() instead of
+                // setProvidedProducts() to populate the providedProducts from the
+                // PoolDTO because the former was marked as @JsonProperty("providedProducts").
+                entity.setProvidedProductDtos(products);
```

Comment:
I don't believe we use the provided product DTOs on deserialization at all. We may be able to ignore these entirely.

Reply:
@Ceiu We do use the providedProducts and derivedProvidedProducts fields on deserialization (at least there are many spec tests that use them). (But not the providedProductDtos/derivedProvidedProductDtos fields).